source/server/http/admin.cc

```diff
@@ -394,26 +395,37 @@ Http::Code AdminImpl::handlerStats(absl::string_view url, Http::HeaderMap& respo
   Http::Code rc = Http::Code::OK;
   const Http::Utility::QueryParams params = Http::Utility::parseQueryString(url);
   std::map<std::string, uint64_t> all_stats;
+  std::map<std::string, std::string> all_histograms;
   for (const Stats::CounterSharedPtr& counter : server_.stats().counters()) {
     all_stats.emplace(counter->name(), counter->value());
   }
 
   for (const Stats::GaugeSharedPtr& gauge : server_.stats().gauges()) {
     all_stats.emplace(gauge->name(), gauge->value());
   }
+  const std::list<Stats::ParentHistogramSharedPtr>& histograms = server_.stats().histograms();
```

Comment:
technically you are storing a reference to a temporary here. Just copy. The compiler will use RVO.